rust-runtime/smithy-http/src/event_stream.rs

```diff
@@ -115,6 +117,15 @@ where
                         .write_to(&mut buffer)
                         .map_err(|err| SdkError::ConstructionFailure(Box::new(err)))?;
                     Poll::Ready(Some(Ok(Bytes::from(buffer))))
+                } else if !*this.end_signal_sent {
+                    *this.end_signal_sent = true;
+                    let mut buffer = Vec::new();
```

Comment:
nit: could we get a size hint from the message? Or does `.write_to` already call reserve?